src/cmds/init/initialize.ts

```diff
@@ -174,25 +175,25 @@ async function buildWeb3Wallet(argv) {
 
   let invalidPermissions = true;
   while (invalidPermissions) {
-    initialPermissions = await prompt(`initialPermissions: [perm1 perm2 ...] ([alert])`);
+    let initialPermissions = (await prompt(`initialPermissions: [perm1 perm2 ...] ([alert])`));
     let err;
     try {
       if (!initialPermissions) {
-        initialPermissions = defaultPerms;
+        initialPermissions = defaultPerms as unknown as string;
         break;
       }
       const splitPermissions = initialPermissions.split(' ')
         .reduce((acc, p) => {
           console.log(p);
           if (typeof p === 'string' && p.match(/^[\w\d_]+$/u)) {
-            acc[p] = {};
+            (acc as any)[p] = {};
           } else {
             logWarning(`Invalid permissions: ${p}`);
           }
           return acc;
         }, {});
 
-      initialPermissions = splitPermissions || defaultPerms;
+      finalPermissions = splitPermissions as any;
```

Comment:
I think we should create an interface for the package.json `initialPermissions` object. That would go a long way to resolve the casting issues in this file. Something like:

```typescript
interface ManifestPermissions {
  [permissionName: string]: {} // just a plain object for now
}
```

Reply:
So the root of this casting here and https://github.com/MetaMask/snaps-cli/pull/75#discussion_r568013489 is actually the fact that prompt returns Promise<string> and then we are processing the permissions as if they are a string. I don't think adding an interface would help because the issue lies on the right side of the colon, which here is still an object but were treating like a string.